### PR TITLE
Basic config map generator function

### PR DESCRIPTION
**Pull request by johnbelamaric**

### CODE REVIEW

**Comment by henderiw** [comment]
could we also add negative tests? like no key provided.

**Comment by johnbelamaric** [comment]
Example of usage: https://github.com/nephio-project/nephio-packages/pull/31

**Comment by vjayaramrh** [comment]
Any instructions on what steps can be tested out on a laptop or a container? Thanks

**Comment by johnbelamaric** [comment]
> Any instructions on what steps can be tested out on a laptop or a container? Thanks

There is an image built and published to the Nephio Docker Hub. This PR makes use of it: https://github.com/nephio-project/nephio-packages/pull/31

You could copy what is done in there or in the unit tests in this PR, and execute the function using `kpt`.

File: krm-functions/gen-configmap-fn/fn/gen_configmap.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		name = rl.FunctionConfig.GetName()
+	}
+
+	err = fc.Validate()
+	if err != nil {
+		return false, err
+	}
+
+	cmko := fn.NewEmptyKubeObject()
+	err = cmko.SetAPIVersion("v1")
```
Comment (efiacor) [inline krm-functions/gen-configmap-fn/fn/gen_configmap.go:91]: Is the version likely to change any time soon?

**Review by efiacor — COMMENTED** [review]
Minor comments

File: krm-functions/gen-configmap-fn/fn/gen_configmap.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+limitations under the License.
+*/
+
+package fn
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+	//"github.com/google/safetext/yamltemplate"
```
[the remark this answers is not in the document]
Comment (johnbelamaric) [reply krm-functions/gen-configmap-fn/fn/gen_configmap.go:23]: yeah, probably. I think I was planning to have a yaml rendering option that would use this library, which protects against yaml injection. but right now the function is ignorant of what format the generated text is in

File: krm-functions/gen-configmap-fn/fn/gen_configmap.go (hunk 1 above)
Comment (johnbelamaric) [reply krm-functions/gen-configmap-fn/fn/gen_configmap.go:91]: No

Comment (johnbelamaric) [reply krm-functions/gen-configmap-fn/fn/gen_configmap.go:91]: (and even if a ConfigMap v2 came out, which would be several years from now, the v1 would still be valid)

File: krm-functions/gen-configmap-fn/fn/gen_configmap.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (efiacor) [reply krm-functions/gen-configmap-fn/fn/gen_configmap.go]: Yep, thought so. Could leave it as a bread crumb for later so.

**Pull request merged**